pkg/kubectl/run.go

```diff
@@ -64,7 +66,25 @@ func makePodSpec(params map[string]string, name string) (*api.PodSpec, error) {
 	return &spec, nil
 }
 
-func (BasicReplicationController) Generate(params map[string]string) (runtime.Object, error) {
+func (BasicReplicationController) Generate(genericParams map[string]interface{}) (runtime.Object, error) {
```

Comment:
This method didn't mutate the `params` before and I wouldn't expect it to based on it's name.  Can you prevent the mutation (preferred) or document it?